src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -324,28 +296,15 @@ public virtual void Write(long value)
         [CLSCompliant(false)]
         public virtual void Write(ulong value)
         {
-            _buffer[0] = (byte)value;
-            _buffer[1] = (byte)(value >> 8);
-            _buffer[2] = (byte)(value >> 16);
-            _buffer[3] = (byte)(value >> 24);
-            _buffer[4] = (byte)(value >> 32);
-            _buffer[5] = (byte)(value >> 40);
-            _buffer[6] = (byte)(value >> 48);
-            _buffer[7] = (byte)(value >> 56);
-            OutStream.Write(_buffer, 0, 8);
+            Write(unchecked((long) value));
         }
 
         // Writes a float to this stream. The current position of the stream is
         // advanced by four.
         //
-        public virtual unsafe void Write(float value)
-        {
-            uint TmpValue = *(uint*)&value;
-            _buffer[0] = (byte)TmpValue;
-            _buffer[1] = (byte)(TmpValue >> 8);
-            _buffer[2] = (byte)(TmpValue >> 16);
-            _buffer[3] = (byte)(TmpValue >> 24);
-            OutStream.Write(_buffer, 0, 4);
+        public virtual void Write(float value)
+        {
+            Write(BitConverter.SingleToInt32Bits(value));
```

Comment:
Note that the `Write(int)` method is virtual. If somebody overwrote the `Write(int)` method, but not `Write(float)` method, they will see a behavior change.